### PR TITLE
Add option to extension's settings: "Trim extra spaces between words"

### LINKED ISSUE #187: Comments placed at end of line shifting left
I often add a comment at the end of a line with four spaces before the semi-colon, and I'm finding that the formatter is concatenating the spaces into a single space.

### Unformatted input

```
#noEnv    ;// prevents empty variables from being looked up
drawMenuItem__exit()    ;// call this function
```

### Actual formatted output

```
#noEnv ;// prevents empty variables from being looked up
drawMenuItem__exit() ;// call this function
```

### PR DESCRIPTION
**Pull request by kyklish**
Closes #187.

Changes proposed in this pull request:

-   make trimming extra spaces between words optional

---

Notifying @mark-wiemer

### CODE REVIEW

File: src/test/suite/providers/formatting/formattingProvider.utils.test.ts [hunk 1]
```diff
@@ -74,4 +75,42 @@ suite('FormattingProvider utils', () => {
...
+            {
+                in: 'InputFile    :=    "movie.mkv"',
+                rs: 'InputFile := "movie.mkv"',
+                ts: true,
+            },
+            {
+                in: 'InputFile    :=    "movie.mkv"',
+                rs: 'InputFile    :=    "movie.mkv"',
+                ts: false,
+            },
```
Comment (mark-wiemer) [inline src/test/suite/providers/formatting/formattingProvider.utils.test.ts:96]:
```suggestion
            },
            {
                in: 'MsgBox,  4,   , testing   testing',
                rs: 'MsgBox, 4, , testing testing',
                ts: true,
            },
```

Comment (mark-wiemer) [reply src/test/suite/providers/formatting/formattingProvider.utils.test.ts:96]: Just want to test something other than assignment here

**Review by mark-wiemer — CHANGES_REQUESTED** [review]
Almost ready to merge!

File: src/providers/formattingProvider.utils.ts [hunk 2]
```diff
@@ -17,3 +17,13 @@ export function hasMoreOpenParens(line: string): boolean {
...
     return openCount > closeCount;
 }
+
+/** @return string with trimmed extra spaces between words*/
+export function trimExtraSpaces(
+    line: string,
+    trimExtraSpaces: boolean,
+): string {
+    return trimExtraSpaces
+        ? line.replace(/ {2,}/g, ' ') // Remove extra spaces between words
```
[the remark this answers is not in the document]
Comment (kyklish) [reply src/providers/formattingProvider.utils.ts:27]: I got this bug (formatter modify quoted string and continuation sections) in my TODO list. It's not easy to fix it here, separate PR needed.

[the remark this answers is not in the document]
Comment (kyklish) [reply src/providers/formattingProvider.utils.ts:27]: Created #199.

[the remark this answers is not in the document]
Comment (mark-wiemer) [reply src/providers/formattingProvider.utils.ts:27]: Awesome, resolving this one.

**Review by mark-wiemer — APPROVED** [review]

**Comment by mark-wiemer** [comment]
Let's resolve merge conflicts and then we're good to go, @kyklish :)

**Comment by kyklish** [comment]
> Let's resolve merge conflicts and then we're good to go, @kyklish :)

Done.

**Pull request merged**